services/alertpost/service.go

```diff
@@ -0,0 +1,155 @@
+package alertpost
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/influxdata/kapacitor/alert"
+	"github.com/influxdata/kapacitor/bufpool"
+	"github.com/influxdata/kapacitor/models"
+)
+
+type Service struct {
+	mu        sync.RWMutex
+	endpoints map[string]Config
+	logger    *log.Logger
+}
+
+func NewService(c Configs, l *log.Logger) *Service {
+	s := &Service{
+		logger:    l,
+		endpoints: c.index(),
+	}
+	return s
+}
+
+type HandlerConfig struct {
+	URL      string `mapstructure:"url"`
+	Endpoint string `mapstructure:"endpoint"`
+}
+
+type handler struct {
+	s        *Service
+	bp       *bufpool.Pool
+	url      string
+	endpoint string
+	logger   *log.Logger
+}
+
+func (s *Service) Handler(c HandlerConfig, l *log.Logger) alert.Handler {
+	return &handler{
+		s:        s,
+		bp:       bufpool.New(),
+		url:      c.URL,
+		endpoint: c.Endpoint,
+		logger:   l,
+	}
+}
+
+func (s *Service) Open() error {
+	return nil
+}
+
+func (s *Service) Close() error {
+	return nil
+}
+
+func (s *Service) endpoint(e string) (c Config, ok bool) {
+	s.mu.RLock()
+	c, ok = s.endpoints[e]
+	s.mu.RUnlock()
+	return
+}
+
+func (s *Service) Update(newConfigs []interface{}) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	for _, nc := range newConfigs {
+		if c, ok := nc.(Config); ok {
+			s.endpoints[c.Endpoint] = c
+		} else {
+			return fmt.Errorf("unexpected config object type, got %T exp %T", nc, c)
+		}
+	}
+	return nil
+}
+
+func (s *Service) Test(options interface{}) error {
+	return nil
+}
+
+type testOptions struct{}
+
+func (s *Service) TestOptions() interface{} {
+	return &testOptions{}
+}
```

Comment:
No, it needs to implement the interface. Also would having and endpoint option make sense so you can test each configured endpoint?